pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -584,22 +587,19 @@ func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeNam
 		}
 		vm, err := vs.getVMByName(ctx, nodeName)
 		if err != nil {
+			if vclib.IsNotFound(err) {
+				glog.Warningf("Node %q does not exist, vsphere CP will assume all disks %v are not attached to it.", nodeName, volPaths)
+				// make all the disks as detached and return false without error.
+				attached := make(map[string]bool)
+				for _, volPath := range volPaths {
+					attached[volPath] = false
+				}
+				return attached, nil
+			}
 			glog.Errorf("Failed to get VM object for node: %q. err: +%v", vSphereInstance, err)
 			return nil, err
 		}
-		nodeExist, err := vm.Exists(ctx)
```

Comment:
@bacongobbler Have you see the comment from @divyenpatel:"Same as commented for DisksAreAttached. When node VM is powered off, we can still check if disks are attached to the Node VM, and return actual state of the disk."
It seem it doesn't matter if the node is powered off or not when the DisksAreAttached() call is made.
On the other hand, the "vm.Exists(ctx)" just check if vm is power on, can't check if vm is exist.